### PR TITLE
Add replies

### LINKED ISSUE #39: As a user, I should be able to reply to posts
This is blocked by sections, because you create posts underneath sections.

### PR DESCRIPTION
**Pull request by risu-kun**
Resolves #39.

This branch makes it possible to reply to posts--I added a form to the post detail page, but that page still needs to be styled the way we want it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -19,16 +19,28 @@ def create
...
-      redirect_to posts_path
+      redirect_to categories_path
+    end
+  end
+
+  def reply
+    @post = Post.find_by_id(params[:id])
+
+    if @post.nil?
+      redirect_to categories_path
```
Comment (kenmazaika) [inline app/controllers/posts_controller.rb]: This should probably be something like this to 404:

```
render text: 'not found', status: :not_found
```

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -73,9 +74,37 @@
...
+
+      expect(response).to redirect_to new_user_session_path
+    end
+
+    it "should redirect to categories#index if the parent post is not found" do
+      user = FactoryGirl.create(:user)
+      sign_in user
+      post :reply, id: "not a real id", post: {message: "Nice post"}
+
+      expect(response).to redirect_to categories_path
```
Comment (kenmazaika) [inline spec/controllers/posts_controller_spec.rb]: This test should be updated as described above.

```
  expect(response).to have_http_status(:not_found)
```

**Comment by kenmazaika** [comment]
Looks good.  Had a couple small nitpicks, but overall :+1:

**Comment by risu-kun** [comment]
Made the changes you suggested, @kenmazaika

Thanks for the feedback :grinning:

**Comment by kenmazaika** [comment]
Looks great!

**Pull request merged**